lib/matplotlib/backend_bases.py

```diff
@@ -2298,32 +2297,32 @@ def get_window_title(self):
             return self.manager.get_window_title()
 
     def set_window_title(self, title):
-        """
-        Set the title text of the window containing the figure.  Note that
-        this has no effect if there is no window (e.g., a PS backend).
+        """Set the title text of the window containing the figure.
+
+        This has no effect if there is no window (e.g., a PS backend).
         """
         if hasattr(self, "manager"):
             self.manager.set_window_title(title)
 
     def get_default_filename(self):
-        """
-        Return a string, which includes extension, suitable for use as
-        a default filename.
-        """
-        default_basename = self.get_window_title() or 'image'
-        default_basename = default_basename.replace(' ', '_')
+        """Return a suitable default filename, including the extension.
+        """
+        default_basename = self.get_window_title() or "image"
+        # Characters to be avoided in a NT path:
+        # https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#naming_conventions
+        removed_chars = \
+            {"posix": " /\0", "nt": r'<>:"/\|?*\0'}.get(os.name, "_")
```

Comment:
Yes.  This is intentional, because there's literally *nothing* wrong with spaces in filenames on Windows (for many users, most relevant files live in "C:\Documents and Settings\...").  (I would argue that it's also the case on Linux but at least you could say well they interact poorly with badly written shell scripts.)

On the other hand I should have mentioned this in the PR doc, sorry about the omission.

I'm going to self-block the PR until we can agree about that change (or until you can convince me to change it back...).